packages/amplify-graphql-transformer-core/src/transform-host.ts

```diff
@@ -134,50 +139,88 @@ export class DefaultTransformHost implements TransformHostProvider {
+    name: string,
+    codeMappingTemplate: MappingTemplateProvider,
+    dataSourceName: string,
+    scope?: Construct,
+    runtime?: CfnFunctionConfiguration.AppSyncRuntimeProperty,
+  ): AppSyncFunctionConfiguration => {
+    return this.addAppSyncFunction(name, { codeMappingTemplate }, dataSourceName, scope, runtime);
+  };
+
+  public addAppSyncVtlRuntimeFunction = (
     name: string,
     requestMappingTemplate: MappingTemplateProvider,
     responseMappingTemplate: MappingTemplateProvider,
     dataSourceName: string,
     scope?: Construct,
     runtime?: CfnFunctionConfiguration.AppSyncRuntimeProperty,
+  ): AppSyncFunctionConfiguration => {
+    return this.addAppSyncFunction(name, { requestMappingTemplate, responseMappingTemplate }, dataSourceName, scope, runtime);
+  };
+
+  public addAppSyncFunction = (
+    name: string,
+    mappingTemplate: FunctionRuntimeTemplate,
+    dataSourceName: string,
+    scope?: Construct,
+    runtime?: CfnFunctionConfiguration.AppSyncRuntimeProperty,
   ): AppSyncFunctionConfiguration => {
     if (dataSourceName && !Token.isUnresolved(dataSourceName) && !this.dataSources.has(dataSourceName)) {
       throw new Error(`DataSource ${dataSourceName} is missing in the API`);
     }
 
     // calculate hash of the slot object
     // if the slot exists for the hash, then return same fn else create function
-
     const dataSource = this.dataSources.get(dataSourceName);
-
+    const hashes = this.getMappingTemplateHash(mappingTemplate);
     const obj: Slot = {
       dataSource: dataSourceName,
-      requestMappingTemplate: requestMappingTemplate.getTemplateHash(),
-      responseMappingTemplate: responseMappingTemplate.getTemplateHash(),
+      ...hashes,
     };
 
     const slotHash = hash(obj);
     if (!this.api.disableResolverDeduping && this.appsyncFunctions.has(slotHash)) {
       const appsyncFunction = this.appsyncFunctions.get(slotHash)!;
       // generating duplicate appsync functions vtl files to help in custom overrides
-      requestMappingTemplate.bind(appsyncFunction, this.api.assetProvider);
-      responseMappingTemplate.bind(appsyncFunction, this.api.assetProvider);
+      this.bindMappingTemplate(mappingTemplate, appsyncFunction, this.api.assetProvider, runtime);
       return appsyncFunction;
     }
 
     const fn = new AppSyncFunctionConfiguration(scope || this.api, name, {
       api: this.api,
       dataSource: dataSource || dataSourceName,
-      requestMappingTemplate,
-      responseMappingTemplate,
+      mappingTemplate,
       runtime,
     });
     this.appsyncFunctions.set(slotHash, fn);
     return fn;
   };
 
-  public addResolver = (
+  public addJsRuntimeResolver = (
+    typeName: string,
+    fieldName: string,
+    codeMappingTemplate: MappingTemplateProvider,
+    resolverLogicalId?: string,
+    dataSourceName?: string,
+    pipelineConfig?: string[],
+    scope?: Construct,
+    runtime?: CfnFunctionConfiguration.AppSyncRuntimeProperty,
```

Comment:
If we're disambiguating the runtime by introducing new runtime specific methods, do we still need this argument?

Reply:
Good call, thanks. I removed them from the `TransformHostProvider` functions already, but missed them here. Addressed in 27f4ea151210acf0bdde352e52068be27b83ecc7